### PR TITLE
Port to 1.20.4

### LINKED ISSUE #90: 1.20.4 incompatibility
On 1.20.4, the game crashes before startup with the smooth swapping mod.

<details>
  <summary>Crash Log</summary>
  
  ```accesslog
  ---- Minecraft Crash Report ----
// There are four lights!

Time: 2023-12-29 10:38:59
Description: Initializing game

java.lang.ExceptionInInitializerError
	at net.minecraft.class_315.<init>(class_315.java:97)
	at net.minecraft.class_310.<init>(class_310.java:489)
	at net.minecraft.client.main.Main.main(Main.java:223)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_339 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at net.minecraft.class_7172.<clinit>(class_7172.java:35)
	... 9 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422)
	... 14 more
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [smoothswapping-common.mixins.json:ClickableWidgetAccessor from mod smoothswapping] from phase [DEFAULT] in config [smoothswapping-common.mixins.json] FAILED during APPLY
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:638)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinApplyError(MixinProcessor.java:589)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:379)
	... 17 more
Caused by: org.spongepowered.asm.mixin.gen.throwables.InvalidAccessorException: No candidates were found matching field_41097:J in net/minecraft/class_339 for smoothswapping-common.mixins.json:ClickableWidgetAccessor from mod smoothswapping->@Accessor[FIELD_SETTER]::setLastHoveredTime(J)V [INJECT Applicator Phase -> smoothswapping-common.mixins.json:ClickableWidgetAccessor from mod smoothswapping -> Apply Accessors ->  -> Locate -> smoothswapping-common.mixins.json:ClickableWidgetAccessor from mod smoothswapping->@Accessor[FIELD_SETTER]::setLastHoveredTime(J)V]
	at org.spongepowered.asm.mixin.gen.AccessorInfo.findTarget(AccessorInfo.java:519)
	at org.spongepowered.asm.mixin.gen.AccessorInfo.findTargetField(AccessorInfo.java:502)
	at org.spongepowered.asm.mixin.gen.AccessorInfo.locate(AccessorInfo.java:476)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.generateAccessors(MixinTargetContext.java:1411)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyAccessors(MixinApplicatorStandard.java:1071)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:327)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	... 17 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.class_315.<init>(class_315.java:97)
	at net.minecraft.class_310.<init>(class_310.java:489)
Mixins in Stacktrace:
	net.minecraft.class_315:
		net.fabricmc.fabric.mixin.client.keybinding.GameOptionsMixin (fabric-key-binding-api-v1.mixins.json)
		net.fabricmc.fabric.mixin.resource.loader.client.GameOptionsMixin (fabric-resource-loader-v0.client.mixins.json)
		com.aizistral.nochatreports.common.mixins.client.MixinOptions (mixins/common/nochatreports.mixins.json)
		de.siphalor.amecs.impl.mixin.MixinGameOptions (amecsapi.mixins.json)
	net.minecraft.class_310:
		net.fabricmc.fabric.mixin.screen.MinecraftClientMixin (fabric-screen-api-v1.mixins.json)
		com.aizistral.nochatreports.common.mixins.client.MixinMinecraft (mixins/common/nochatreports.mixins.json)
		dev.isxander.yacl3.mixin.MinecraftMixin (yacl.mixins.json)
		dev.isxander.zoomify.mixins.hooks.MinecraftClientMixin (zoomify.mixins.json)
		net.fabricmc.fabric.mixin.event.interaction.client.MinecraftClientMixin (fabric-events-interaction-v0.client.mixins.json)
		de.siphalor.mousewheelie.client.mixin.MixinMinecraftClient (mousewheelie.mixins.json)
		de.siphalor.amecs.impl.mixin.MixinMinecraftClient (amecsapi.mixins.json)
		com.aizistral.nochatreports.fabric.mixins.client.MixinMinecraft (mixins/fabric/nochatreports-fabric.mixins.json)
		net.fabricmc.fabric.mixin.datagen.client.MinecraftClientMixin (fabric-data-generation-api-v1.client.mixins.json)
		me.jellysquid.mods.sodium.mixin.core.MinecraftClientMixin (sodium.mixins.json)
		net.fabricmc.fabric.mixin.networking.client.accessor.MinecraftClientAccessor (fabric-networking-api-v1.client.mixins.json)
		net.fabricmc.fabric.mixin.registry.sync.client.MinecraftClientMixin (fabric-registry-sync-v0.client.mixins.json)
		dynamic_fps.impl.mixin.MinecraftMixin (dynamic_fps.mixins.json)
		net.fabricmc.fabric.mixin.event.lifecycle.client.MinecraftClientMixin (fabric-lifecycle-events-v1.client.mixins.json)

-- Initialization --
Details:
	Modules: 
		ADVAPI32.dll:Advanced Windows 32 Base API:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		COMCTL32.dll:User Experience Controls Library:6.10 (WinBuild.160101.0800):Microsoft Corporation
		CRYPT32.dll:Crypto API32:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTBASE.dll:Base cryptographic API DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTSP.dll:Cryptographic Service Provider API:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		DBGHELP.DLL:Windows Image Helper:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		DNSAPI.dll:DNS Client API DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		GDI32.dll:GDI Client DLL:10.0.22621.2792 (WinBuild.160101.0800):Microsoft Corporation
		IMM32.DLL:Multi-User Windows IMM32 API Client DLL:10.0.22621.2792 (WinBuild.160101.0800):Microsoft Corporation
		IPHLPAPI.DLL:IP Helper API:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		KERNEL32.DLL:Windows NT BASE API Client DLL:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		KERNELBASE.dll:Windows NT BASE API Client DLL:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		MpOav.dll:IOfficeAntiVirus Module:4.18.23110.3 (9ebb3643d539a6fc4659898b1df3124d5da4c0a9):Microsoft Corporation
		NSI.dll:NSI User-mode interface DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		NTASN1.dll:Microsoft ASN.1 API:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		OLEAUT32.dll:OLEAUT32.DLL:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		Ole32.dll:Microsoft OLE for Windows:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		PSAPI.DLL:Process Status Helper:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		Pdh.dll:Windows Performance Data Helper DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		RPCRT4.dll:Remote Procedure Call Runtime:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		SHCORE.dll:SHCORE:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		SHELL32.dll:Windows Shell Common Dll:10.0.22621.2792 (WinBuild.160101.0800):Microsoft Corporation
		USER32.dll:Multi-User Windows USER API Client DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		USERENV.dll:Userenv:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		VCRUNTIME140.dll:Microsoft® C Runtime Library:14.31.31103.0:Microsoft Corporation
		VERSION.dll:Version Checking and File Installation Libraries:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		WINHTTP.dll:Windows HTTP Services:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		WINMM.dll:MCI API DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		WS2_32.dll:Windows Socket 2.0 32-Bit DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		WSOCK32.dll:Windows Socket 32-Bit DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		amsi.dll:Anti-Malware Scan Interface:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		bcrypt.dll:Windows Cryptographic Primitives Library:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		bcryptPrimitives.dll:Windows Cryptographic Primitives Library:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		clbcatq.dll:COM+ Configuration Catalog:2001.12.10941.16384 (WinBuild.160101.0800):Microsoft Corporation
		combase.dll:Microsoft COM for Windows:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		dbgcore.DLL:Windows Core Debugging Helpers:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc.DLL:DHCP Client Service:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc6.DLL:DHCPv6 Client:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		fwpuclnt.dll:FWP/IPsec User-Mode API:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		gdi32full.dll:GDI Client DLL:10.0.22621.2861 (WinBuild.160101.0800):Microsoft Corporation
		glfw.dll:GLFW 3.4.0 DLL:3.4.0:GLFW
		java.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		javaw.exe:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		jemalloc.dll
		jimage.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		jli.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		jna17677129438174444410.dll:JNA native library:6.1.6:Java(TM) Native Access (JNA)
		jsvml.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		jvm.dll:Java HotSpot(TM) 64-Bit server VM:17.0.9.0:Oracle Corporation
		kernel.appcore.dll:AppModel API Host:10.0.22621.2715 (WinBuild.160101.0800):Microsoft Corporation
		lwjgl.dll
		management.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		management_ext.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		msvcp140.dll:Microsoft® C Runtime Library:14.31.31103.0:Microsoft Corporation
		msvcp_win.dll:Microsoft® C Runtime Library:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		msvcrt.dll:Windows NT CRT DLL:7.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		mswsock.dll:Microsoft Windows Sockets 2.0 Service Provider:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		napinsp.dll:E-mail Naming Shim Provider:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		ncrypt.dll:Windows NCrypt Router:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		net.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		nio.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		nlansp_c.dll:NLA Namespace Service Provider DLL:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		ntdll.dll:NT Layer DLL:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		perfos.dll:Windows System Performance Objects DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		pfclient.dll:SysMain Client:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		pnrpnsp.dll:PNRP Name Space Provider:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		profapi.dll:User Profile Basic API:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		rasadhlp.dll:Remote Access AutoDial Helper:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		rsaenh.dll:Microsoft Enhanced Cryptographic Provider:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		sechost.dll:Host for SCM/SDDL/LSA Lookup APIs:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		shlwapi.dll:Shell Light-weight Utility Library:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		sunmscapi.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		ucrtbase.dll:Microsoft® C Runtime Library:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		vcruntime140_1.dll:Microsoft® C Runtime Library:14.31.31103.0:Microsoft Corporation
		verify.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		win32u.dll:Win32u:10.0.22621.2861 (WinBuild.160101.0800):Microsoft Corporation
		windows.storage.dll:Microsoft WinRT Storage API:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		winrnr.dll:LDAP RnR Provider DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		wintypes.dll:Windows Base Types DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		wshbth.dll:Windows Sockets Helper DLL:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		wshunix.dll:AF_UNIX Winsock2 Helper DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		zip.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:223)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Mixins in Stacktrace: None found

-- System Details --
Details:
	Minecraft Version: 1.20.4
	Minecraft Version ID: 1.20.4
	Operating System: Windows 11 (amd64) version 10.0
	Java Version: 17.0.9, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
	Memory: 184095360 bytes (175 MiB) / 549453824 bytes (524 MiB) up to 4294967296 bytes (4096 MiB)
	CPUs: 24
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 9 7900X 12-Core Processor            
	Identifier: AuthenticAMD Family 25 Model 97 Stepping 2
	Microarchitecture: Zen 3
	Frequency (GHz): 4.70
	Number of physical packages: 1
	Number of physical CPUs: 12
	Number of logical CPUs: 24
	Graphics card #0 name: NVIDIA GeForce RTX 4070
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x2786
	Graphics card #0 versionInfo: DriverVersion=31.0.15.4601
	Memory slot #0 capacity (MB): 16384.00
	Memory slot #0 clockSpeed (GHz): 4.80
	Memory slot #0 type: Unknown
	Memory slot #1 capacity (MB): 16384.00
	Memory slot #1 clockSpeed (GHz): 4.80
	Memory slot #1 type: Unknown
	Virtual memory max (MB): 52925.93
	Virtual memory used (MB): 38506.29
	Swap memory total (MB): 20480.00
	Swap memory used (MB): 131.66
	JVM Flags: 3 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms512m -Xmx4096m
	Fabric Mods: 
		betterf3: BetterF3 9.0.0
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.11.15+78d798af4f
		borderlessmining: Borderless Mining 1.1.9+1.20.2
		capes: Capes 1.5.3+1.20.2
		cloth-config: Cloth Config v13 13.0.121
			cloth-basic-math: cloth-basic-math 0.6.1
		dynamic_fps: Dynamic FPS 3.3.2
		entityculling: EntityCulling 1.6.3.1
		fabric-api: Fabric API 0.92.0+1.20.4
			fabric-api-base: Fabric API Base 0.4.35+78d798af4f
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.45+78d798af4f
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.15+78d798af4f
			fabric-block-api-v1: Fabric Block API (v1) 1.0.14+78d798af4f
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.3+78d798af4f
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.45+78d798af4f
			fabric-client-tags-api-v1: Fabric Client Tags 1.1.6+78d798af4f
			fabric-command-api-v1: Fabric Command API (v1) 1.2.40+f71b366f4f
			fabric-command-api-v2: Fabric Command API (v2) 2.2.19+78d798af4f
			fabric-commands-v0: Fabric Commands (v0) 0.2.57+df3654b34f
			fabric-containers-v0: Fabric Containers (v0) 0.1.81+df3654b34f
			fabric-content-registries-v0: Fabric Content Registries (v0) 5.0.9+78d798af4f
			fabric-convention-tags-v1: Fabric Convention Tags 1.5.9+78d798af4f
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.22+78d798af4f
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 13.1.17+78d798af4f
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.60+78d798af4f
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.5.28+4ced05924f
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.7.0+389931eb4f
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.71+df3654b34f
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.45+78d798af4f
			fabric-item-api-v1: Fabric Item API (v1) 2.1.34+78d798af4f
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.0.20+78d798af4f
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.40+78d798af4f
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.38+df3654b34f
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.29+78d798af4f
			fabric-loot-api-v2: Fabric Loot API (v2) 2.1.4+78d798af4f
			fabric-message-api-v1: Fabric Message API (v1) 6.0.4+78d798af4f
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.59+78d798af4f
			fabric-model-loading-api-v1: Fabric Model Loading API (v1) 1.0.7+78d798af4f
			fabric-models-v0: Fabric Models (v0) 0.4.6+9386d8a74f
			fabric-networking-api-v1: Fabric Networking API (v1) 3.1.2+78d798af4f
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 13.0.8+06274a474f
			fabric-particles-v1: Fabric Particles (v1) 1.1.6+78d798af4f
			fabric-recipe-api-v1: Fabric Recipe API (v1) 2.0.15+78d798af4f
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 4.0.13+78d798af4f
			fabric-renderer-api-v1: Fabric Renderer API (v1) 3.2.3+78d798af4f
			fabric-renderer-indigo: Fabric Renderer - Indigo 1.5.3+78d798af4f
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.50+df3654b34f
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.41+73761d2e4f
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.32+78d798af4f
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.53+df3654b34f
			fabric-rendering-v1: Fabric Rendering (v1) 3.0.12+78d798af4f
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.3.13+78d798af4f
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.16+78d798af4f
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.50+78d798af4f
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.16+78d798af4f
			fabric-transfer-api-v1: Fabric Transfer API (v1) 4.0.6+78d798af4f
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 5.0.13+78d798af4f
		fabric-language-kotlin: Fabric Language Kotlin 1.10.17+kotlin.1.9.22
			org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.9.22
			org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.9.22
			org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.9.22
			org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.9.22
			org_jetbrains_kotlinx_atomicfu-jvm: atomicfu-jvm 0.23.1
			org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.7.3
			org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.7.3
			org_jetbrains_kotlinx_kotlinx-datetime-jvm: kotlinx-datetime-jvm 0.5.0
			org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.6.2
			org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.6.2
			org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.6.2
		fabricloader: Fabric Loader 0.15.3
			mixinextras: MixinExtras 0.3.2
		java: Java HotSpot(TM) 64-Bit Server VM 17
		minecraft: Minecraft 1.20.4
		mixintrace: MixinTrace 1.1.1+1.17
		modmenu: Mod Menu 9.0.0
		mousewheelie: Mouse Wheelie 1.13.3+mc1.20.4
			amecsapi: Amecs API 1.5.6+mc1.20.2
			coat: Coat 1.0.0-beta.22+mc1.20.4
			tweed4_annotated: tweed4_annotated 1.3.1+mc1.20.2
			tweed4_base: tweed4_base 1.7.1+mc1.20.2
			tweed4_data: tweed4_data 1.2.1+mc1.20.2
			tweed4_data_hjson: tweed4_data_hjson 1.1.1+mc1.20.2
			tweed4_tailor_coat: tweed4_tailor_coat 1.1.3+mc1.20.2
			tweed4_tailor_lang_json_descriptions: tweed4_tailor_lang_json_descriptions 1.1.0+mc1.20.2
			tweed4_tailor_screen: tweed4_tailor_screen 1.1.4+mc1.20.2
		nochatreports: No Chat Reports 1.20.4-v2.5.0
		smoothswapping: Smooth Swapping 0.9.3.1
		sodium: Sodium 0.5.5
		starlight: Starlight 1.1.3+fabric.f5dcd1a
		yet_another_config_lib_v3: YetAnotherConfigLib 3.3.1+1.20.4
			com_twelvemonkeys_common_common-image: common-image 3.10.0
			com_twelvemonkeys_common_common-io: common-io 3.10.0
			com_twelvemonkeys_common_common-lang: common-lang 3.10.0
			com_twelvemonkeys_imageio_imageio-core: imageio-core 3.10.0
			com_twelvemonkeys_imageio_imageio-metadata: imageio-metadata 3.10.0
			com_twelvemonkeys_imageio_imageio-webp: imageio-webp 3.10.0
			org_quiltmc_parsers_gson: gson 0.2.1
			org_quiltmc_parsers_json: json 0.2.1
		zoomify: Zoomify 2.12.0
			com_akuleshov7_ktoml-core-jvm: ktoml-core-jvm 0.5.0
			dev_isxander_settxi_settxi-core: settxi-core 2.10.6
			dev_isxander_settxi_settxi-kotlinx-serialization: settxi-kotlinx-serialization 2.10.6
	Launched Version: 1.20.4
	Launcher name: PrismLauncher
	Backend library: LWJGL version 3.3.2-snapshot
	Backend API: Unknown
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Universe: 404
	Type: Client (map_client.txt)
	Locale: en_US
	CPU: <unknown>
  ```
  
</details>

### PR DESCRIPTION
**Pull request by dima-dencep**
Fixes #90

### CODE REVIEW

**Comment by PixelTails** [comment]
Yes, update to 1.20.4

**Review by PixelTails — APPROVED** [review]
Yes, update to 1.20.4

**Comment by ViTulyankin** [comment]
@dima-dencep can you please publish a ready-to-use jar or explain how to build it by myself?

File: gradle.properties [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
-
 #Fabric
-fabric_loader_version=0.14.22
-fabric_api_version=0.89.0+1.20.1
-modmenu=7.2.2
+fabric_loader_version=0.15.3
+fabric_api_version=0.93.1+1.20.4
+modmenu=9.0.0
 
 #NeoFroge
```
Comment (notlin4) [inline gradle.properties:16]: Typo
```suggestion
#NeoForge
```

Comment (dima-dencep) [reply gradle.properties:16]: Note that I did not change this line.

**Comment by Enderteck** [comment]
Will compatibilty come soon ?

**Comment by Olexandr43** [comment]
Here is 1.20.4 port: https://modrinth.com/mod/smooth-swapping-1.20.4-port

**Comment by dima-dencep** [comment]
I will never send pr to any mod again, enjoy it
Too bad for the author of this mod for having his work used by such people.

**Comment by Enderteck** [comment]
> I will never send pr to any mod again, enjoy it Too bad for the author of this mod for having his work used by such people.

Why, what happened ?
Has Olexandr43 done somethign bad ?
What is stopping you from sending prs to mods ?

**Comment by JasonHorkles** [comment]
I second that. Seems a bit odd to close the PR just to open another one to "get rid of the spam". It's not like it changes visibility at all

**Comment by Enderteck** [comment]
> I second that. Seems a bit odd to close the PR just to open another one to "get rid of the spam". It's not like it changes visibility at all

Especially because it's not being merged lol. 

Why complain about the spam if the pull request won't do anything...

We can talk about it.

**Pull request closed without merging**